### PR TITLE
Hide `disputed on` column in disputes listing

### LINKED ISSUE #6661: Remove `Disputed on` date column or hide by default
### Description

Remove the `Disputed on` column from the Disputes listing.

image

No changes should be made to the CSV export.

### Test Instructions

1. Visit the disputes listing page, `Payments > Disputes`
2. The `Disputed on` column should not be visible.

### PR DESCRIPTION
**Pull request by brucealdridge**
Fixes #6661

#### Changes proposed in this Pull Request

Hide `Disputed on` column in the disputes listing.

The changes in #6355 require removing the `disputed on` column. As the CSV is generated using the same code path, removing the column wasn't a straightforward option. The merchant is still able to add the `Disputed on` column back into the report.

The default sort order of the dispute listing is by the `Disputed on` in descending order. As we are hiding this column now it may make sense to update the default sort order to the `Respond by` column. 

image

#### Testing instructions

1. Visit the Disputes listing page, Payments > Disputes
2. See that the `Disputed On` column is not visible.
3. Download the csv and see that the results are unchanged.
4. Enable the column by using the options menu.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6666` or branch name `fix/6661-hide-disputed-on-column` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=fix/6661-hide-disputed-on-column) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: af0c6e5cee97589eb0f0a5a6a8a2b32b9d46e4f8
- Build time: 2023-07-06 03:00:56 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: changelog/fix-6661-hide-disputed-on-column [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Significance: patch
+Type: update
+Comment: Hide disputed on date column, part of a larger set of changes for disputes
```
Comment (shendy-a8c) [inline changelog/fix-6661-hide-disputed-on-column:3]: I think this change worths to be mentioned in public changelog, or do you plan to add a changelog entry in other PRs?

**Review by shendy-a8c — CHANGES_REQUESTED** [review]
Nice PR number. Diff looks good. Tested well. Nice comment addition to make it clear what the array index represents in the CSV.

I have feedback about changelog.

> The default sort order of the dispute listing is by the Disputed on in descending order. As we are hiding this column now it may make sense to update the default sort order to the Respond by column.

Also, I agree about this. I think the default sort should be by the `Respond by` column. That change is not yet part of this PR, right?

**Comment by brucealdridge** [comment]
> I think the default sort should be by the `Respond by` column.

I've added this as a task on the Epic. I noticed that @haszari [suggested this](https://github.com/Automattic/woocommerce-payments/issues/6355#issuecomment-1558454473) as well. I think it makes sense to include as part of this PR.

File: changelog/fix-6661-hide-disputed-on-column (hunk 1 above)
Comment (brucealdridge) [reply changelog/fix-6661-hide-disputed-on-column:3]: I plan to do as part of the Epic. To ensure this doesn't get missed, I've added this as a task on the Epic.

**Comment by brucealdridge** [comment]
> I think it makes sense to include as part of this PR.

I've had a look into this and it is slightly more complex and makes sense to be part of highlighting urgent dates or separately. I've created #6692 

Ordering this column by `DESC` will show the latest due date first, whereas ordering by `ASC` will show disputes that have already been responded to first. As the `Respond by` date remains after the due date.

**Review by shendy-a8c — APPROVED** [review]
Since using other column as default sort will be done in https://github.com/Automattic/woocommerce-payments/issues/6692, this PR looks good.

**Pull request merged**